cedar-policy/src/tests.rs

```diff
@@ -1709,6 +1710,158 @@ mod entity_validate_tests {
             }
         }
     }
+
+    /// Record inside entity doesn't conform to schema
```

Comment:
Bonus points for tests where a (required/optional) inner record attribute is _missing_ in the entity data.  (as opposed to this test where there is an _extra_ attribute in the entity data.)  But these tests adequately address the #1176 report.

Reply:
Latest rev tests look good